src/test/java/mate/academy/security/CustomUserDetailsServiceTest.java

```diff
@@ -0,0 +1,54 @@
+package mate.academy.security;
+
+import java.util.Optional;
+import java.util.Set;
+import mate.academy.model.Role;
+import mate.academy.model.User;
+import mate.academy.service.UserService;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+
+class CustomUserDetailsServiceTest {
+    private static final String EMAIL = "bob@i.ua";
+    private static final String PASSWORD = "1234";
+    private static final String WRONG_EMAIL = "abc@i.ua";
+    private UserDetailsService userDetailsService;
+    private UserService userService;
+    private User bob;
+
+    @BeforeEach
+    void setUp() {
+        userService = Mockito.mock(UserService.class);
+        userDetailsService = new CustomUserDetailsService(userService);
+        bob = new User();
+        bob.setEmail(EMAIL);
+        bob.setPassword(PASSWORD);
+        bob.setRoles(Set.of(new Role(Role.RoleName.USER)));
+    }
+
+    @Test
+    void loadUserByUsername_Ok() {
+        Mockito.when(userService.findByEmail(EMAIL)).thenReturn(Optional.of(bob));
+        UserDetails actual = userDetailsService.loadUserByUsername(EMAIL);
+        Assertions.assertNotNull(actual);
+        Assertions.assertEquals(EMAIL, actual.getUsername());
+        Assertions.assertEquals(PASSWORD, actual.getPassword());
+    }
+
+    @Test
+    void loadUserByUsername_UserNameNotFound() {
+        Mockito.when(userService.findByEmail(EMAIL)).thenReturn(Optional.of(bob));
+        try {
+            userDetailsService.loadUserByUsername(WRONG_EMAIL);
+        } catch (UsernameNotFoundException e) {
+            Assertions.assertEquals("User not found.", e.getMessage());
+            return;
+        }
```

Comment:
You shouldn't use try catch in tests, here, we can do this instead:
```suggestion
UsernameNotFoundException exception = Assertions.assertThrows(
    UsernameNotFoundException.class,
    () -> userDetailsService.loadUserByUsername(WRONG_EMAIL)
);

Assertions.assertEquals("User not found.", exception.getMessage());
```